### PR TITLE
Fix panics discovered by Go-Fuzz and update tests

### PR DESCRIPTION
**Pull request by parsiya**
- Fixed crashes found by Go-Fuzz.
- Added crashes as tests.
- Updated one of the original tests that does not expect a large tag value error.

### CODE REVIEW

**Review by xor-gate — COMMENTED** [review]
When this is not necessary anymore we should remove it.

File: exif/exif_test.go [hunk 1]
```diff
@@ -205,9 +205,9 @@ func TestHugeTagError(t *testing.T) {
 	if err == nil {
 		t.Fatal("no error on bad exif data")
 	}
-	if !strings.Contains(err.Error(), "short read") {
-		t.Fatal("wrong error:", err.Error())
-	}
+	// if !strings.Contains(err.Error(), "short read") {
```
Comment (xor-gate) [inline exif/exif_test.go:208]: When this is not necessary anymore we should remove it.

File: exif/gofuzz_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"os"
+	"testing"
+)
+
+var goFuzzPayloads = make(map[string]string)
+
+// Populate payloads.
+func populatePayloads() {
+
+	goFuzzPayloads["3F"] = "II*\x00\b\x00\x00\x00\t\x000000000000" +
```
Comment (xor-gate) [inline exif/gofuzz_test.go:14]: Me and probably other people could not be familiar with gofuzz, maybe you should write a small paragraph the library could be tested with gofuzz?

**Comment by xor-gate** [comment]
ping @parsiya could we finish the few review comments? then we can merge 👍

File: exif/gofuzz_test.go (hunk 2 above)
Comment (parsiya) [reply exif/gofuzz_test.go:14]: Sorry I am on vacation so I cannot do any sort of meaningful work for the next 3 weeks or so. If you want to go ahead and do something without my notes please do.

As for introduction, I wrote a couple of blog posts about go-fuzz. This one is about the crash I reported: https://parsiya.net/blog/2018-05-05-learning-go-fuzz-2-goexif2/ (there's another one mentioned in that blog post).

**Comment by xor-gate** [comment]
ping @parsiya ?

**Pull request closed without merging**